ampersand-chart.js

```diff
@@ -613,12 +632,27 @@
       if (this.model.drawXAxisLabels) {
         container.append('text')
           .attr('class', 'ampersand-graph-label')
+          .style('display', 'none')
+          .style('opacity', 0)
           .attr('y', height - 46)
           .attr('dy', '1.25em');
 
-        containers.select('text.ampersand-graph-label')
-          .transition()
-          .attr('x', ((barWidth * values.length) + barMargin * (values.length - 1)) / 2);
+        _.defer(function(containers, barWidth, values, barMargin, label) {
+          containers.select('text.ampersand-graph-label')
+            .style('opacity', 0)
+            .style('display', function(d, i) {
+              var width = Math.ceil(_.max(data, function(item) { return item[label].toString().length; })[label].toString().length * 6.25);
+              var barGroupWidth = (barWidth * values.length) + barMargin * (values.length - 1);
+              if (width < barGroupWidth || this.model.drawAllXAxisLabels) {
+                return undefined;
+              }
+
+              return i % Math.ceil(width / barGroupWidth) === 0 ? undefined : 'none';
+            }.bind(this))
+            .transition()
+            .style('opacity', 1)
+            .attr('x', ((barWidth * values.length) + barMargin * (values.length - 1)) / 2);
```

Comment:
you repeat this calculation for barGroupWidth in here.  It could definitely be broken out (in this local case, just a single variable at the top of the _.defer is enough, but it may make sense to split it out further - I haven't looked at the rest of the code to see if this is elsewhere.